server/user.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * @fileoverview user function handler
+ */
+var fs = require('fs');
+var assert = require('assert');
+
+var log = require('./log.js');
+var utils = require('./utils.js');
+
+/** @type {user} */
+module.exports = user;
+
+/**
+ * @constructor
+ */
+function user() {}
+
+/** @enum {string} */
+user.QueryType = {
+  SIGNUP: 'sign-up',
+  SIGNIN: 'sign-in'
+};
+
+/**
+ * Name of user information file.
+ * @type {string}
+ */
+user.userInfoFile = 'userInfo.txt';
+
+/**
+ * Expire time of cookie.
+ * @const {number}
+ */
+user.cookieExpireTime = 24 * 60 * 60 * 1000;
+
+/**
+ * @typedef {{
+ *   email: (string|undefined),
+ *   username: string,
+ *   password: string,
+ *   confirmed: (boolean|undefined)
+ * }}
+ */
+user.userInfo;
+
+/**
+ * @typedef {{
+ *   error: (Object<{
+ *     type: string,
+ *     message: string
+ *   }>)
+ * }}
+ */
+user.Error;
+
+/**
+ * Checks the user information for signing in.
+ * @param {string} userPath Directory of user indormation file.
+ * @param {!user.userInfo} userInfo User Information
+ * @return {user.Error|boolean}
+ */
+user.signUp = function(userPath, userInfo) {
+  var checkDuplicate = {
+    duplicated: false,
+    elements: []
+  };
+  var lines = fs.readFileSync(userPath + user.userInfoFile).toString()
+    .split('\n');
+  for (var i = 0; i < lines.length; i++) {
+    var parts = lines[i].split(/[\t\s]+/);
+    if (parts.length == 0) {
+      continue;
+    }
+    if (parts[0] == userInfo.email || parts[1] == userInfo.username) {
+      checkDuplicate.duplicated = true;
+      if (parts[0] == userInfo.email) {
+        checkDuplicate.elements.push('email: ' + userInfo.email);
+      }
+      if (parts[1] == userInfo.username) {
+        checkDuplicate.elements.push('username: ' + userInfo.username);
+      }
+      break;
+    }
+  }
+  if (checkDuplicate.duplicated) {
+    var errorMessage = checkDuplicate.elements.join(' and ') + ' exist';
+    errorMessage += checkDuplicate.elements.length == 1 ? 's' : '';
+    return {
+      error: {
+        type: 'sign-up',
+        message: errorMessage
+      }
+    };
+  } else {
+    var infoLine = userInfo.email + ' ' + userInfo.username + ' ' +
```

Comment:
BTW it may be better to write
```js
[
  userInfo.email,
  userInfo.username,
  userInfo.password,
  userInfo.confirmed
].join(' ') + '\n';
```
If you know the order of keys in the object, you can also do
```js
_.values(userInfo).join(' ') + '\n'
```
though after switching to mongodb I don't think you will keep this line.